packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -506,6 +507,7 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
   @override
   void initState() {
     super.initState();
+    _focusNode = widget.focusNode ?? FocusNode();
```

Comment:
We should follow the pattern where we have:
a) A user-provided FocusNode
b) An internally managed FocusNode, if the user never provided one

So you might want to have an "_effectiveFocudeNode" getter for the other parts of this change.

Reply:
done